### ISSUE TITLE
Error with shared-places-list

### ISSUE THREAD

**Issue by LarsRabe** [vesta-webtrees-2-custom-modules/vesta_shared_places#142]
Goto Lists > Vesta Shared Places, 
Click one of the entries, 
The following error is thrown

```Class "Cissee\Webtrees\Module\PersonalFacts\DefaultAssociateFactUtils" not found …/modules_v4/vesta_shared_places/resources/views/shared-place-page-details.phtml:33
#0 …/app/View.php(183): include()
#1 …/app/View.php(278): Fisharebest\Webtrees\View->render()
#2 …/app/Helpers/functions.php(145): Fisharebest\Webtrees\View::make()
#3 …/modules_v4/vesta_shared_places/resources/views/shared-place-page.phtml(55): view()
#4 …/app/View.php(183): include('...')
#5 …/app/View.php(278): Fisharebest\Webtrees\View->render()
#6 …/app/Helpers/functions.php(145): Fisharebest\Webtrees\View::make()
#7 …/app/Http/ViewResponseTrait.php(50): view()
#8 …/modules_v4/vesta_shared_places/patchedWebtrees/Http/RequestHandlers/SharedPlacePage.php(155): Cissee\WebtreesExt\Http\RequestHandlers\SharedPlacePage->viewResponse()
#9 …/app/Http/Middleware/RequestHandler.php(55): Cissee\WebtreesExt\Http\RequestHandlers\SharedPlacePage->handle()
#10 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\RequestHandler->process()
#11 …/modules_v4/default_language/DefaultLanguageMiddleware.php(106): Middleland\Dispatcher->handle()
#12 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): DefaultLanguageMiddlewareNamespace\DefaultLanguageMiddleware->process()
#13 …/app/Module/CheckForNewVersion.php(122): Middleland\Dispatcher->handle()
#14 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Module\CheckForNewVersion->process()
#15 …/app/Http/Middleware/CheckCsrf.php(80): Middleland\Dispatcher->handle()
#16 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\CheckCsrf->process()
#17 …/vendor/oscarotero/middleland/src/Dispatcher.php(118): Middleland\Dispatcher->handle()
#18 …/app/Webtrees.php(275): Middleland\Dispatcher->dispatch()
#19 …/app/Http/Middleware/Router.php(161): Fisharebest\Webtrees\Webtrees::dispatch()
#20 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\Router->process()
#21 …/app/Http/Middleware/BootModules.php(60): Middleland\Dispatcher->handle()
#22 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\BootModules->process()
#23 …/app/Http/Middleware/RegisterGedcomTags.php(54): Middleland\Dispatcher->handle()
#24 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\RegisterGedcomTags->process()
#25 …/app/Http/Middleware/LoadRoutes.php(75): Middleland\Dispatcher->handle()
#26 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\LoadRoutes->process()
#27 …/app/Http/Middleware/CheckForNewVersion.php(57): Middleland\Dispatcher->handle()
#28 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\CheckForNewVersion->process()
#29 …/app/Http/Middleware/UseTransaction.php(45): Middleland\Dispatcher->handle()
#30 …/vendor/illuminate/database/Concerns/ManagesTransactions.php(29): Fisharebest\Webtrees\Http\Middleware\UseTransaction::Fisharebest\Webtrees\Http\Middleware\{closure}()
#31 …/app/Http/Middleware/UseTransaction.php(46): Illuminate\Database\Connection->transaction()
#32 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseTransaction->process()
#33 …/app/Http/Middleware/DoHousekeeping.php(73): Middleland\Dispatcher->handle()
#34 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\DoHousekeeping->process()
#35 …/app/Http/Middleware/UseTheme.php(69): Middleland\Dispatcher->handle()
#36 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseTheme->process()
#37 …/app/Http/Middleware/CheckForMaintenanceMode.php(51): Middleland\Dispatcher->handle()
#38 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\CheckForMaintenanceMode->process()
#39 …/app/Http/Middleware/UseLanguage.php(71): Middleland\Dispatcher->handle()
#40 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseLanguage->process()
#41 …/app/Http/Middleware/UseSession.php(78): Middleland\Dispatcher->handle()
#42 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseSession->process()
#43 …/app/Http/Middleware/UpdateDatabaseSchema.php(57): Middleland\Dispatcher->handle()
#44 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UpdateDatabaseSchema->process()
#45 …/app/Http/Middleware/UseDatabase.php(122): Middleland\Dispatcher->handle()
#46 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseDatabase->process()
#47 …/app/Http/Middleware/BadBotBlocker.php(279): Middleland\Dispatcher->handle()
#48 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\BadBotBlocker->process()
#49 …/app/Http/Middleware/CompressResponse.php(73): Middleland\Dispatcher->handle()
#50 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\CompressResponse->process()
#51 …/app/Http/Middleware/ContentLength.php(40): Middleland\Dispatcher->handle()
#52 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\ContentLength->process()
#53 …/vendor/middlewares/client-ip/src/ClientIp.php(65): Middleland\Dispatcher->handle()
#54 …/app/Http/Middleware/ClientIp.php(47): Middlewares\ClientIp->process()
#55 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\ClientIp->process()
#56 …/app/Http/Middleware/HandleExceptions.php(90): Middleland\Dispatcher->handle()
#57 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\HandleExceptions->process()
#58 …/app/Http/Middleware/BaseUrl.php(73): Middleland\Dispatcher->handle()
#59 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\BaseUrl->process()
#60 …/app/Http/Middleware/ReadConfigIni.php(68): Middleland\Dispatcher->handle()
#61 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\ReadConfigIni->process()
#62 …/app/Http/Middleware/SecurityHeaders.php(48): Middleland\Dispatcher->handle()
#63 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\SecurityHeaders->process()
#64 …/app/Http/Middleware/EmitResponse.php(57): Middleland\Dispatcher->handle()
#65 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\EmitResponse->process()
#66 …/vendor/oscarotero/middleland/src/Dispatcher.php(118): Middleland\Dispatcher->handle()
#67 …/app/Webtrees.php(275): Middleland\Dispatcher->dispatch()
#68 …/app/Webtrees.php(262): Fisharebest\Webtrees\Webtrees::dispatch()
#69 …/index.php(51): Fisharebest\Webtrees\Webtrees->httpRequest()
#70 {main}```

**Comment by ric2016** [comment]
I cannot reproduce this.
You pasted the stack trace - what is the actual error message (if any)?

Which webtrees and php (and Vesta) version?

**Comment by LarsRabe** [comment]
no error-message
webtrees = latest > 2.1.15
vesta = latest > 2.1.15.0.1 for common and 2.1.15.0.0 for the rest
php = 8.1.13

try https://www.ravenzwaaij.info/tree/ravenzwaaij/shared-place-list
and https://www.ravenzwaaij.info/tree/ravenzwaaij/sharedPlace/X336/Veenendaal

**Comment by ric2016** [comment]
So the error is 
```
Class "Cissee\Webtrees\Module\PersonalFacts\DefaultAssociateFactUtils" not found …\modules_v4\vesta_shared_places\resources\views\shared-place-page-details.phtml:33
```



<img width="853" alt="error" src="https://user-images.githubusercontent.com/18174787/211193399-dc3b7fc2-7abf-47a8-91b4-e18eecafec77.png">

(by the way, if this first line does not show in your browser, this should be investigated separately)

I can reproduce this now. So there is a dependency from the shared places module to another Vesta module, that's not good (code should be in Vesta-Common in that case). I'll have to re-organize the code. 

In the meantime as a workaround, the error should disappear if you add the folder "vesta_personal_facts" to the "modules_v4" directory (shouldn't be necessary to enable the module)

**Comment by LarsRabe** [comment]
You're correct. I didn't install `vesta_personal_facts` as for now I do not want to use it. I just found the depency myself and will use the workaround until you fix it.

BTW: the first line is showing now after I disabled all other custom modules. I'll try around to see which one of theme is responsible for that.

THX Lars

**Comment by ric2016** [comment]
Should be fixed via the referenced commit. Vesta release to follow.

Thanks for ivestigating! I'll leave this issue open for now in case something comes up re: In which cases the error line gets printed, that should be useful for similar issues.

**Comment by LarsRabe** [comment]
Wow, you're "_faster as the fire brigade_" (to use a german saying). The patch works without the workaround now.

I've found that the 2.0.0-beta3 version of `jc-fancy-treeview` with all the patches after the beta-release was the cause of the not showing the error. After I updated to the official 2.0.0-release the error is shown. The only patch I was missing there, was for this issue: https://github.com/JustCarmen/webtrees-fancy-treeview/issues/114
So likely that was somehow involved.
